### ISSUE TITLE
setDataSource failed.: status=0x80000000 on android kitkat

### ISSUE THREAD

**Issue by odahan** [adrianstevens/Xamarin-Plugins#5]
There is a problem on android 4 with music path. This error is known and someone has found the solution (in Java but it's similar in c#).
This problem is making the app crashing.
Can you apply the correction to avoid crash on Android 4.x (still about 25% of users) ?
Thanks in advance !
And thanks for this really useful plugin too :-)

the thread where you'll find the description of the problem and its solution  : 
https://stackoverflow.com/questions/16395559/mediaplayer-setdatasource-failed-with-status-0x80000000-for-ringtone-set-by-file

**Comment by adrianstevens** [comment]
thx again @odahan 

I'll wrap the SetDataSource call in a try block (which I should have done anyways) and I'll fall back to the overload that accepts the context and Uri.

Something like this:
~~~
try
{
    player.SetDataSource(path);
}
catch
{
   try
   {
      player?.SetDataSource(context, Uri.Parse(Uri.Encode(path)));
    }
   catch
    {
       ... 
    }
}
~~~

**Comment by adrianstevens** [comment]
This should be fixed as of version 0.9.0. Closing the issue. I've tested on two physical devices and the emulator. Closing for now, please re-open if you see any crashes.
